app/services/people/importer.rb

```diff
@@ -2,60 +2,89 @@
 
 module People
   class Importer
-    PROFILE_IDENTIFIERS_MAPPING = {
+    NOMIS_IDENTIFIER_MAPPING = {
       'police_national_computer' => :pnc_number,
       'prison_number' => :prison_number,
       'criminal_records_office' => :cro_number,
     }.freeze
 
-    attr_accessor :nomis_attributes
-
     def initialize(nomis_attributes)
-      self.nomis_attributes = nomis_attributes
+      @nomis_attributes = nomis_attributes
     end
 
     def call
       person = Person.find_or_initialize_by(nomis_prison_number: nomis_attributes[:prison_number])
+      person.assign_attributes(person_attributes)
+
       profile = person.latest_profile || person.profiles.build
-      profile.assign_attributes(attributes)
-      # return profile rather than person so that save does both profile and person if both are new
+      profile.assign_attributes(profile_attributes)
+
       profile
     end
 
   private
 
-    def attributes
+    attr_reader :nomis_attributes
+
+    def person_attributes
+      nomis_attributes.slice(:last_name, :date_of_birth).merge(
+        first_names: first_names,
+        gender: gender,
```

Comment:
we're calling a few of those attributes twice, in person and profile, maybe it's worth memoizing those values in the methods below?

Reply:
Done